pkg/apis/autoscaling/validation/declarative_validation_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	apitesting "k8s.io/kubernetes/pkg/api/testing"
+	"k8s.io/kubernetes/pkg/apis/autoscaling"
+	"k8s.io/kubernetes/pkg/features"
+)
+
+func TestValidateScaleForDeclarative(t *testing.T) {
+	ctx := genericapirequest.WithRequestInfo(genericapirequest.NewDefaultContext(), &genericapirequest.RequestInfo{
+		APIGroup:   "autoscaling",
+		APIVersion: "v1",
+	})
+	testCases := map[string]struct {
+		input        autoscaling.Scale
+		expectedErrs field.ErrorList
+	}{
+		// spec.replicas
+		"0 replicas": {
+			input: mkScale(setScaleSpecReplicas(0)),
+		},
+		"1 replicas": {
+			input: mkScale(setScaleSpecReplicas(1)),
+		},
+		"positive replicas": {
+			input: mkScale(setScaleSpecReplicas(100)),
+		},
+		"negative replicas": {
+			input: mkScale(setScaleSpecReplicas(-1)),
+			expectedErrs: field.ErrorList{
+				field.Invalid(field.NewPath("spec.replicas"), nil, "").WithOrigin("minimum"),
+			},
+		},
+	}
+	for k, tc := range testCases {
+		t.Run(k, func(t *testing.T) {
+			var declarativeTakeoverErrs field.ErrorList
+			var imperativeErrs field.ErrorList
+			for _, gateVal := range []bool{true, false} {
+				// We only need to test both gate enabled and disabled together, because
```

Comment:
I suggest adding:

```
t.Run(fmt.Sprintf("gates=%v", gateVal), func(t *testing.T) {
```

So any test fails are easier to read (from debugging the cohorts PR)

Reply:
Adding. We'll need to patch up https://github.com/kubernetes/kubernetes/blob/30469e180361d7da07b0fee6d47c776fa2cf3e86/pkg/registry/core/replicationcontroller/declarative_validation_test.go#L34 as well.  I'll follow up.

Reply:
That's in my cohorts PR